src/Api/Request/ParametersBuilder.php

```diff
@@ -0,0 +1,141 @@
+     */
+    protected $logger;
+
+    public function __construct( LoggerFactory $loggerFactory)
+    {
+        $this->logger = $loggerFactory->getLogger(Plugin::PLUGIN_NAMESPACE, Plugin::PLUGIN_IDENTIFIER);
+    }
+
+    /**
+     * @return CategoryService|null
+     */
+    public function getCategoryService()
+    {
+        if (!$this->categoryService) {
+            $this->categoryService = pluginApp(CategoryService::class);
+        }
+
+        return $this->categoryService;
+    }
+
+    /**
+     * @param Request $request
+     * @param HttpRequest $httpRequest
+     * @param Category|null $category
+     * @return Request
+     */
+    public function setSearchParams($request, $httpRequest, $category = null)
+    {
+        $parameters = $httpRequest->all();
+
+        $request->setParam('query', $parameters['query'] ?? '');
+        $request->setPropertyParam(Plugin::API_PROPERTY_MAIN_VARIATION_ID);
+
+        if (isset($parameters[Plugin::API_PARAMETER_ATTRIBUTES])) {
+            $attributes = $parameters[Plugin::API_PARAMETER_ATTRIBUTES];
+            foreach ($attributes as $key => $value) {
+                if ($key === 'cat' && $category) {
+                    continue;
+                }
+
+                $request->setAttributeParam($key, $value);
+            }
+        }
+
+        if ($category && ($categoryFullName = $this->getCategoryName($category))) {
+            $request->setParam('selected', ['cat' => [$categoryFullName]]);
+            //TODO: remove after testing
+            $this->logger->error('Category full name: ' . $categoryFullName);
+        }
+
+        if (isset($parameters[Plugin::API_PARAMETER_SORT_ORDER]) && in_array($parameters[Plugin::API_PARAMETER_SORT_ORDER], Plugin::API_SORT_ORDER_AVAILABLE_OPTIONS)) {
+            $request->setParam(Plugin::API_PARAMETER_SORT_ORDER, $parameters[Plugin::API_PARAMETER_SORT_ORDER]);
+        }
+
+        $request = $this->setPagination($request, $parameters);
+
+        return $request;
+    }
+
+    /**
+     * @param Request $request
+     * @param array $parameters
+     * @return Request
+     */
+    public function setPagination($request, $parameters)
+    {
+        $pageSize = $parameters[Plugin::API_PARAMETER_PAGINATION_ITEMS_PER_PAGE] ?? 0;
+
+        if (intval($pageSize) > 0) {
+            $request->setParam(Plugin::API_PARAMETER_PAGINATION_ITEMS_PER_PAGE, $pageSize);
+        }
+
+        $paginationStart = $parameters[Plugin::API_PARAMETER_PAGINATION_START] ?? 0;
+
+        if (intval($paginationStart) > 0) {
+            $request->setParam(Plugin::API_PARAMETER_PAGINATION_START, $paginationStart);
+        }
+
+        return $request;
+    }
+
+    /**
+     * @param Category $category
+     * @return string
+     */
+    public function getCategoryName($category)
+    {
+        $categoryName = '';
+
+        try {
+            $categoryTree = $this->getCategoryTree($category);
+            array_reverse($categoryTree);
```

Comment:
This doesn't seem to fix the issue since I'm still getting the categories in the wrong order. Also the respective test implies that the wrong order is returned.